script/prerender/manage.ts

```diff
@@ -0,0 +1,397 @@
+      },
+      {
+        ParameterKey: 'ValidUntil',
+        ParameterValue: `${timeoutDate.toISOString().slice(0, -5)}Z`,
+      },
+    ],
+    StackName: WORKERS_STACK_NAME,
+    Tags: [
+      {Key: 'Project', Value: 'Unified'},
+      {Key: 'Application', Value: 'Rex'},
+      {Key: 'Environment', Value: 'shared'},
+      {Key: 'Owner', Value: 'dante'},
+    ],
+    TemplateBody: readFile(path.join(__dirname, 'cfn.yml')),
+  }));
+}
+
+async function deleteWorkersStack() {
+  console.log('Started workers stack deletion');
+
+  return cfnClient.send(new DeleteStackCommand({StackName: WORKERS_STACK_NAME}));
+}
+
+async function prepareAndQueueBook([bookId, {defaultVersion}]: [string, {defaultVersion: string}]) {
+  // Don't have the book title yet at this point
+  console.log(`Loading book ${bookId}@${defaultVersion}`);
+
+  const book = await bookLoader(bookId, defaultVersion);
+
+  console.log(`[${book.title}] Book loaded; preparing pages`);
+
+  const pages = prepareBookPages(book).map((page) => omit('route', page.route));
+  const numBookPages = pages.length;
+  numPages += numBookPages;
+
+  // Wait until the work queue is ready
+  const workQueueUrl = await workQueuePromise;
+
+  console.log(`[${book.title}] Queuing ${numBookPages} book pages in batches of 10`);
+
+  for (let pageIndex = 0; pageIndex < pages.length; pageIndex += 10) {
+    const pageBatch = pages.slice(pageIndex, pageIndex + 10);
+
+    // If the entire request fails, this command will throw and be caught at the end of this file
+    // However, we also need to check if only some of the messages failed
+    const sendMessageBatchResult = await sqsClient.send(new SendMessageBatchCommand({
+      Entries: pageBatch.map((page, batchIndex) => ({
+        Id: batchIndex.toString(),
+        MessageBody: JSON.stringify({ payload: { page }, type: 'page' }),
+      })),
+      QueueUrl: workQueueUrl,
+    }));
+
+    const failedMessages = sendMessageBatchResult.Failed || [];
+    const numFailures = failedMessages.length;
+    if (numFailures > 0) {
+      throw new Error(`[SQS] [SendMessageBatch] Error: ${numFailures} out of ${pageBatch.length
+        } pages in a batch failed to be queued. Failures: ${JSON.stringify(failedMessages)}`);
+    }
+
+    const successfulMessages = sendMessageBatchResult.Successful || [];
+    const numSuccesses = successfulMessages.length;
+    if (numSuccesses < pageBatch.length) {
+      throw new Error(`[SQS] [SendMessageBatch] Unexpected response: received only ${numSuccesses
+        } successes out of ${pageBatch.length} pages in a batch but also no failures. Successes: ${
+        JSON.stringify(successfulMessages)}`);
+    }
+  }
+
+  console.log(`[${book.title}] All ${numBookPages} book pages queued`);
+
+  // console.log(`[${book.title}] Queuing sitemap`);
+
+  await sqsClient.send(new SendMessageCommand({
+    MessageBody: JSON.stringify({ payload: { pages, slug: book.slug }, type: 'sitemap' }),
```

Comment:
you don't have any typing for these queue bodies, it might be helpful to create some aliases for encoding them and some guards that match the type on read so that you can get strong typing

Reply:
I did already add the guards on read, but I'll add the types as well.

Reply:
I added types when sending messages. You may want to check the receiving side too though. I have some types there, but it mostly relies on runtime checks.